framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Domain/Repositories/EntityFrameworkCore/EfCoreRepository.cs

```diff
@@ -37,30 +35,50 @@ async Task<DbContext> IEfCoreRepository<TEntity>.GetDbContextAsync()
     [Obsolete("Use GetDbContextAsync() method.")]
     private TDbContext GetDbContext()
     {
+        TDbContext dbContext;
         // Multi-tenancy unaware entities should always use the host connection string
         if (!EntityHelper.IsMultiTenant<TEntity>())
         {
             using (CurrentTenant.Change(null))
             {
-                return _dbContextProvider.GetDbContext();
+                dbContext = _dbContextProvider.GetDbContext();
             }
         }
+        else
+        {
+            dbContext = _dbContextProvider.GetDbContext();
+        }
+
+        if (IsReadOnly)
+        {
+            dbContext.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;
```

Comment:
Different repositories share the same `DbContext` object in a unit of work. Setting `QueryTrackingBehavior` will effect all other repositories. So, do not make that.